### PR TITLE
Boost: remove wp-content/advanced-cache.php when Cache module is disabled

### PR DESCRIPTION
**Pull request by donnchawp**
When the Cache module is disabled we should remove the file wp-content/advanced-cache.php so that other caching plugins can be easily used alongside Boost.

ref: [#37227](https://github.com/Automattic/jetpack/issues/37227)

## Proposed changes:

* In Page_Cache.php I added code to handle_module_status_updated() that deletes the advanced-cache.php if the module "page_cache" was disabled.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pc9hqz-2QH-p2

## Testing instructions:
* Apply PR
* Enable the Cache module and confirm the file wp-content/advanced-cache.php was created.
* Disable the Cache module and check that wp-content/advanced-cache.php was deleted.

### CODE REVIEW

File: projects/plugins/boost/app/modules/optimizations/page-cache/Page_Cache.php [hunk 1]
```diff
@@ -69,6 +69,10 @@ public function handle_module_status_updated( $module_slug, $status ) {
 		if ( in_array( $module_slug, $slugs, true ) ) {
 			$this->invalidate_cache();
 		}
+
+		if ( $module_slug === 'page_cache' && ! $status ) {
+			Page_Cache_Setup::delete_advanced_cache();
+		}
```
Comment (haqadn) [inline projects/plugins/boost/app/modules/optimizations/page-cache/Page_Cache.php]: This works nicely! However, as uncle Bob said, one function should do a single thing, and this function seems to be doing two things now. 

I suggest we split the functionality in two different functions. Something like:
- `clear_cache_on_output_changing_module_toggle()`
- `delete_advanced_cache`

**Review by haqadn — CHANGES_REQUESTED** [review]

File: projects/plugins/boost/app/modules/optimizations/page-cache/Page_Cache.php (hunk 1 above)
Comment (donnchawp) [reply projects/plugins/boost/app/modules/optimizations/page-cache/Page_Cache.php]: Good idea, done!

**Review by dilirity — APPROVED** [review]
Works as expected. Nice work!

I see you've addressed @haqadn's feedback.

:shipit:

**Review by haqadn — APPROVED** [review]
👍🏼

**Pull request merged**